addons/web/controllers/main.py

```diff
@@ -424,15 +424,22 @@ def xml2json_from_elementtree(el, preserve_whitespaces=False):
     res["children"] = kids
     return res
 
-def binary_content(xmlid=None, model='ir.attachment', id=None, field='datas', unique=False,
-                   filename=None, filename_field='datas_fname', download=False, mimetype=None,
-                   default_mimetype='application/octet-stream', related_id=None, access_mode=None, access_token=None,
-                   env=None):
-    return request.registry['ir.http'].binary_content(
-        xmlid=xmlid, model=model, id=id, field=field, unique=unique, filename=filename,
-        filename_field=filename_field, download=download, mimetype=mimetype,
-        default_mimetype=default_mimetype, related_id=related_id, access_mode=access_mode, access_token=access_token,
-        env=env)
+def limited_image_resize(content, width=None, height=None, crop=False, upper_limit=False, avoid_if_small=False, headers=None):
+    is_svg = headers and dict(headers).get('Content-Type') == 'image/svg+xml'
```

Comment:
if content.startswith  89 50 4E 47 or FF D8 FF E0

Reply:
> if content.startswith 89 50 4E 47 or FF D8 FF E0

That's what we do to sniff the mimetype, let's not duplicate it everywhere.

Reply:
We'll probably move this method to odoo.tools, so we can use it without headers

Reply:
Noe lets hardcode it here,  i dont want to unzip stuff here.